### PR TITLE
Assorted Morsels

### PR DESCRIPTION
**Pull request by ethteck**
Most of the work here was in src/evt - highlights include some major player / npc api funcs as well as main script engine stuff

### CODE REVIEW

File: src/evt/si.c [hunk 1]
```diff
@@ -1175,7 +1178,336 @@ s32 func_802C73B8(ScriptInstance* script) {
     return 1;
 }
 
-INCLUDE_ASM(s32, "evt/si", si_execute_next_command);
+s32 si_execute_next_command(ScriptInstance *script) {
```
Comment (nanaian) [inline src/evt/si.c:1181]: gg

File: src/evt/si.c [hunk 2]
```diff
@@ -1175,7 +1178,336 @@ s32 func_802C73B8(ScriptInstance* script) {
...
+        if (status >= 0) {
+            if (status == ApiStatus_BLOCK) {
+            } else if (status == ApiStatus_DONE1) {
+                script->currentOpcode = 0;
+            } else if (status == ApiStatus_DONE2) {
+                script->currentOpcode = 0;
+                if (gGameStatusPtr->disableScripts != status) {
+                    continue;
+                }
+            } else {
```
Comment (nanaian) [inline src/evt/si.c:1502]: Guess now we know the difference between DONE1 and DONE2!

File: src/evt/si.c [hunk 3]
```diff
@@ -1450,29 +1778,85 @@ INCLUDE_ASM(Bytecode*, "evt/si", si_skip_else, ScriptInstance* script);
...
+    s32* opcode;
+    s32* nargs;
+
+    do {
+        opcode = pos++;
+        nargs = pos++;
+
+        pos += *nargs;
+
+        switch(*opcode) {
```
Comment (nanaian) [inline src/evt/si.c:1831]: format

File: src/evt/si.c [hunk 4]
```diff
@@ -1450,29 +1778,85 @@ INCLUDE_ASM(Bytecode*, "evt/si", si_skip_else, ScriptInstance* script);
...
+            case 23:
+            case 24:
+            case 25:
+            case 26:
+            case 27:
+            case 28:
+            case 29:
+            case 30:
+            case 32:
+            case 33:
```
Comment (nanaian) [inline src/evt/si.c:1853]: Could we use ScriptOpcode_x in the switch statements in this file pls?

File: src/evt/si.c [hunk 5]
```diff
@@ -1450,29 +1778,85 @@ INCLUDE_ASM(Bytecode*, "evt/si", si_skip_else, ScriptInstance* script);
...
         }
     } while(1);
-}*/
+}
 
 INCLUDE_ASM(Bytecode*, "evt/si", si_goto_end_case, ScriptInstance* script);
 
-INCLUDE_ASM(Bytecode*, "evt/si", si_goto_next_case, ScriptInstance* script);
+Bytecode* si_goto_next_case(ScriptInstance* script) {
+    s32 temp = 1;
```
Comment (nanaian) [inline src/evt/si.c:1820]: probably switchDepth

File: src/evt/si.c [hunk 6]
```diff
@@ -1175,7 +1178,336 @@ s32 func_802C73B8(ScriptInstance* script) {
     return 1;
 }
 
-INCLUDE_ASM(s32, "evt/si", si_execute_next_command);
+s32 si_execute_next_command(ScriptInstance *script) {
+    while (TRUE) {
+        s32 status = ApiStatus_DONE2;
+        s32* lines;
+        s32 argC;
```
Comment (nanaian) [inline src/evt/si.c:1185]: Call this nargs to be consistent with funcs lower down in the file? Or change nargs to argc

File: src/world/action/hit_fire.c [hunk 7]
```diff
@@ -1,3 +1,60 @@
 #include "common.h"
 
-INCLUDE_ASM(void, "world/action/hit_fire", func_802B6000_E27C90, void);
+extern f32 D_802B62B0;
+
+//INCLUDE_ASM(void, "world/action/hit_fire", func_802B6000_E27C90, void);
```
Comment (nanaian) [inline src/world/action/hit_fire.c:5]: rm

**Review by nanaian — CHANGES_REQUESTED** [review]

File: src/a5dd0_len_114e0.c [hunk 8]
```diff
@@ -1078,13 +1076,306 @@ INCLUDE_ASM(s32, "a5dd0_len_114e0", func_80116698);
...
+//             if (!(modelTransformGroup->flags & 2)) {
+//                 renderTaskPtr->appendGfx = func_8011800C;
+//                 renderTaskPtr->appendGfxArg = modelTransformGroup;
+//                 renderTaskPtr->distance = -distance;
+//                 renderTaskPtr->renderMode = modelTransformGroup->renderMode;
+//                 queue_render_task(renderTaskPtr);
+//             }
+//         }
+//     }
+// }
```
[the remark this answers is not in the document]
Comment (ethteck) [reply src/a5dd0_len_114e0.c:1370]: ah yeah I forgot about this. It's not really that close atm. I can remove the comment if you prefer

**Pull request merged**